File: libs/gomodels/channel_category_alias.go
package model

type ChannelCategoryAlias struct {
	ID        string `gorm:"primary_key;AUTO_INCREMENT;column:id;type:UUID;"  json:"id"`
	ChannelID string `gorm:"column:channelId;type:TEXT;"  json:"channelId"`
	Category  string `gorm:"column:category;type:TEXT;"  json:"category"`
	Alias     string `gorm:"column:alias;type:TEXT;"  json:"alias"`
}

func (c *ChannelCategoryAlias) TableName() string {
	return "channels_categories_aliases"
}

File: libs/typeorm/src/entities/ChannelCategoryAlias.ts
import {
	Column,
	Entity,
	Index,
	JoinColumn,
	ManyToOne,
	PrimaryGeneratedColumn,
	Unique,
} from 'typeorm';

// eslint-disable-next-line import/no-cycle
import { Channel } from './Channel';

@Entity('channels_categories_aliases', { schema: 'public' })
@Unique(['alias', 'channelId'])
export class ChannelCategoryAlias {
	@PrimaryGeneratedColumn('uuid')
	id: string;

	@Column('text', { name: 'category', nullable: false })
	category: string;

	@Column('text', { name: 'categoryId', nullable: false })
	categoryId: string;

	@Column('text', { name: 'alias', nullable: false })
	alias: string;

	@ManyToOne(() => Channel, (_) => _.categoriesAliases, {
		onDelete: 'RESTRICT',
		onUpdate: 'CASCADE',
	})
	@JoinColumn([{ name: 'channelId', referencedColumnName: 'id' }])
	channel?: Channel;

	@Index()
	@Column('text', { name: 'channelId' })
	channelId: string;
}
